docs/frontend_architecture/HTML5_API.rst

```diff
@@ -0,0 +1,93 @@
+Custom Navigation
+=================
+
+The purpose of the ``kolibri.js`` extension of our HTML5 API is to allow a sandboxed HTML5 app to safely request the main Kolibri application's data.
+
+External/partner product teams can create HTML5 applications that are fully embeddable within Kolibri and can read Kolibri content data, which they otherwise wouldn't be able to access. This opens up possibilities for creative ways in which learners can engage with content, because partners can create any type of app they want. The app could be something completely new, developed for a content source that we are adding to the platform, or it could be a branded, offline recreation of a partner's existing learning app that previously would not have been able to exist on Kolibri.
+
+When a user has permissions to access a custom channel, and they click on it in the main learn tab, rather than viewing "normal Kolibri," they will experience a full-screen HTML5 app. One "out-of-the-box" user interaction is the "navigateTo()" function, which opens  a modal that displays a content node. For other data fetching requests, the app, not Kolibri, has the responsibilty of determining what to do with that data.
+
+
+Basic API
+~~~~~~~~~
+
+
+Access the API from within an HTML5 app by using ``window.kolibri.[function]``
+
+Functions:
+
+.. code-block:: javascript
+
+    /*
+    * Method to query contentnodes from Kolibri and return
+    * an array of matching metadata
+    * @param {Object} options - The different options to filter by
+    * @param {string=} options.parent - id of the parent node to filter by, or 'self'
+    * @param {string[]} options.ids - an array of ids to filter by
+    * @return {Promise<PageResult>} - a Promise that resolves to an array of ContentNodes
+    */
+    getContentByFilter(options)
+
+    /*
+    * Method to query a single contentnode from Kolibri and return
+    * a metadata object
+    * @param {string} id - id of the ContentNode
+    * @return {Promise<ContentNode>} - a Promise that resolves to a ContentNode
+    */
+    getContentById(id)
+
+    /*
+    * Method to search for contentnodes on Kolibri and return
+    * an array of matching metadata
+    * @param {Object} options - The different options to search by
+    * @param {string=} options.keyword - search term for key word search
+    * @param {string=} options.under - id of topic to search under, or 'self'
+    * @return {Promise<PageResult>} - a Promise that resolves to an array of ContentNodes
```

Comment:
Could you add the `typedefs` for `PageResult` and other Object types in the docs as well?